### PR TITLE
[TOPIC-GPIO] revisions to winc1500 driver PR

### PR DESCRIPTION
**Pull request by pabigot**
This PR augments #22156 with additional commits to resolve various problems.

From https://github.com/zephyrproject-rtos/zephyr/pull/22156#pullrequestreview-350271511 are commits to fix the build, improve validation, and test the build.

From https://github.com/zephyrproject-rtos/zephyr/pull/22156#discussion_r372499830 is a change to set an initial output value.

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]

File: drivers/wifi/winc1500/wifi_winc1500.c [hunk 1]
```diff
@@ -1069,25 +1070,35 @@ static int winc1500_init(struct device *dev)
...
 	}
 
-	m2m_wifi_set_scan_region(WINC1500_REGION);
-
 	socketInit();
 	registerSocketCallback(winc1500_socket_cb, NULL);
 
-	m2m_wifi_get_otp_mac_address(w1500_data.mac, &is_valid);
+	if (m2m_wifi_get_otp_mac_address(w1500_data.mac, &is_valid)) {
+		LOG_ERR("Failed set power profile");
```
Comment (jfischer-phytec-iot) [inline drivers/wifi/winc1500/wifi_winc1500.c]: Log message is not correct, my fault, sorry.

Comment (pabigot) [reply drivers/wifi/winc1500/wifi_winc1500.c]: What do you want it to be?  "Failed to set MAC address"?

Comment (pabigot) [reply drivers/wifi/winc1500/wifi_winc1500.c]: I'll update it next push.

Comment (jfischer-phytec-iot) [reply drivers/wifi/winc1500/wifi_winc1500.c]: @pabigot Please change to "Failed to get MAC address", thanks

Comment (pabigot) [reply drivers/wifi/winc1500/wifi_winc1500.c]: done

File: drivers/wifi/winc1500/wifi_winc1500.c [hunk 2]
```diff
@@ -1069,25 +1070,35 @@ static int winc1500_init(struct device *dev)
...
 		return -EIO;
 	}
 
-	m2m_wifi_set_scan_region(WINC1500_REGION);
-
 	socketInit();
 	registerSocketCallback(winc1500_socket_cb, NULL);
 
-	m2m_wifi_get_otp_mac_address(w1500_data.mac, &is_valid);
+	if (m2m_wifi_get_otp_mac_address(w1500_data.mac, &is_valid)) {
```
Comment (mnkp) [inline drivers/wifi/winc1500/wifi_winc1500.c]:
```
if (m2m_wifi_get_otp_mac_address(w1500_data.mac, &is_valid) != M2M_SUCCESS) {
```

Comment (pabigot) [reply drivers/wifi/winc1500/wifi_winc1500.c]: Updated

**Review by mnkp — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**